modules/ardrivo/test/String.cxx

```diff
@@ -0,0 +1,227 @@
+#include <catch2/catch.hpp>
+#include "WString.h"
+
+using Catch::Contains;
+using Catch::EndsWith; 
+
+TEST_CASE("Test compare a string to another" "[compareTo]")
+{
+	String test = "test";
+	String test2 = "test";
+	String test3 = "test";
+	String test4 = "est";	
+	REQUIRE(!test.compareTo(test2));
+	REQUIRE(test3.compareTo(test4));
+}
+
+TEST_CASE("Tests whether or not a String starts with the characters of another String" "[startsWith]")
+{
+	String test = "test";
+	String test2 = "test";
+	String test3 = "est";
+	String test4 = "test";	
+
+	REQUIRE(test.startsWith(test2));
+	REQUIRE(!test3.startsWith(test4));
+}
+
+TEST_CASE("Tests whether or not a String ends with the characters of another String" "[endsWith]")
+{
+	String test = "test";
+	String test2 = "test";
+	String test3 = "est";
+	String test4 = "test";
+
+	REQUIRE(test.startsWith(test2));
+	REQUIRE(!test3.startsWith(test4));
+}
+
+TEST_CASE("Copies the String's characters to the supplied buffer" "[getBytes]")
+{
+	String test = "test";
+	byte arr[5];
+	test.getBytes(arr, test.length());	
+	const char *p = reinterpret_cast<const char*>(arr);
+	REQUIRE_THAT(p, Contains("test"));
+}
+
+TEST_CASE("Locates a character or String within another String" "[indexOf]")
+{
+	String test = "test";
+	String test2 = "test";
+	String test3 = "est";
+
+	REQUIRE(!test.indexOf(test2));
+	REQUIRE(test.indexOf(test3));
```

Comment:
`.indexOf` returns an integer which should probably be checked for exactness rather than narrowed to a `bool`